### PR TITLE
chore: add an example for Deep-SAD-PyTorch [DET-5229]

### PR DESCRIPTION
**Pull request by ioga**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: examples/anomaly_detection/deep_sad/model_def.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
+from pathlib import Path
```
Comment (katport) [inline examples/anomaly_detection/deep_sad/model_def.py:1]: add comments to the top about the original script and highlight of this file

File: examples/anomaly_detection/deep_sad/model_def.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            torch.optim.lr_scheduler.MultiStepLR(
+                self.optimizer, milestones=(50, 100), gamma=0.1
+            ),
+            step_mode=LRScheduler.StepMode.STEP_EVERY_EPOCH,
+        )
+
+        self.download_directory = f"/tmp/data-rank{self.context.distributed.get_rank()}"
+        self.dataset: Optional[TorchvisionDataset] = None
+
+        self.eps = 1e-6
```
Comment (katport) [inline examples/anomaly_detection/deep_sad/model_def.py:173]: set to hparam

File: examples/anomaly_detection/deep_sad/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+If you have not yet installed Determined, installation instructions can be found
+under `docs/install-admin.html` or at https://docs.determined.ai/latest/index.html
+
+You can run the entire example using `./run.sh`. To go more in-depth, you can
+run the individual autoencoder model using  `det e create const_ae.yaml . -f`,
+then download its checkpoint using `det e download <EXPERIMENT_ID>`,
+copy `state_dict.pth` from the downloaded checkpoint to `./ae_state_dict.pth`,
+and run the main model via `det e create const_main.yaml . -f`.
+
+## Results
```
Comment (katport) [inline examples/anomaly_detection/deep_sad/README.md:37]: add expected training time

File: examples/anomaly_detection/deep_sad/model_def.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        outputs = self.model(inputs)
+        dist = torch.sum((outputs - self.c) ** 2, dim=1)
+        losses = torch.where(
+            semi_targets == 0,
+            dist,
+            self.eta * ((dist + self.eps) ** semi_targets.float()),
+        )
+        loss = torch.mean(losses)
+        loss.backward()
+        self.optimizer.step()
```
Comment (katport) [inline examples/anomaly_detection/deep_sad/model_def.py:196]: Is there a reason not to use the contex.backwards here?

File: examples/anomaly_detection/deep_sad/model_def.py (hunk 1 above)
Comment (ioga) [reply examples/anomaly_detection/deep_sad/model_def.py:11]: added a description and links. what kind of highlights do you mean?..

File: examples/anomaly_detection/deep_sad/model_def.py (hunk 4 above)
Comment (ioga) [reply examples/anomaly_detection/deep_sad/model_def.py]: good catch. it's one of the ambiguous porting points. under what circumstances would someone still use `loss.backward` and raw `optimizer.step()` instead of our context way? can/should we warn the user somehow?

**Comment by ioga** [comment]
Moved to model-zoo pending implementation of workflow management / task dependencies / DAGs / pipelines in some form.
Commit: https://github.com/determined-ai/model-zoo/commit/d1f3080678de6c098fbdade8772163a40ff0938e
